scripts/merge-pr.sh

```diff
@@ -15,30 +20,127 @@ confirm() {
   fi
 }
 
-echo "[Merge PR from ${BRANCH}]"
+load_config() {
+  [[ -f merge-pr.conf ]] && . merge-pr.conf
+  : ${OWNER:=status-im}
+  : ${REPO:=status-react}
+  : ${REMOTE:=origin}
+  : ${BRANCH:=develop}
+}
+
+check_pr_prereq() {
+  if ! command -v jq >/dev/null; then
+    fatal "jq(1) is not found, PR cannot be queried."
+  fi
+  if ! command -v curl >/dev/null; then
+    fatal "curl(1) is not found, PR cannot be queried."
+  fi
+}
+
+GH_URL_BASE="https://api.github.com"
+
+get_pr_info() {
+  echo '[ Reading PR info ]'
+  local pr=$1
+  local pr_info_url="$GH_URL_BASE/repos/${OWNER}/${REPO}/pulls/$pr"
+  set +e
+  local pr_info
+  pr_info=$(curl -fsS "$pr_info_url")
+  if [ $? -ne 0 ]; then
+    fatal "Unable to get PR info from $pr_info_url"
+  fi
+  set -e
+  if [[ $(echo "$pr_info" | jq -r .state) == closed ]]; then
+    fatal "PR $pr is closed, will not merge"
+  fi
+  if [[ $(echo "$pr_info" | jq -r .maintainer_can_modify) == true ]]; then
+    RW_PR_REPO=1
+  else
+    warn "PR does not allow 'edits from maintainers', so it will be kept open"
+  fi
+  PR_URL=$(echo "$pr_info" | jq -r .head.repo.ssh_url)
+  PR_REMOTE_NAME=pr-$pr
+  PR_BRANCH=$(echo "$pr_info" | jq -r .head.ref)
+  PR_LOCAL_BRANCH=pr-$pr
+}
+
+fetch_pr() {
+  echo '[ Fetching PR ]'
+  git remote add $PR_REMOTE_NAME $PR_URL
+  git fetch $PR_REMOTE_NAME $PR_BRANCH
+}
+
+refresh_base_branch() {
+  git fetch $REMOTE $BRANCH
+}
+
+rebase_pr() {
+  git checkout -B $PR_LOCAL_BRANCH $PR_REMOTE_NAME/$PR_BRANCH
+  git rebase $BRANCH
+}
+
+check_is_pr_single_commit() {
```

Comment:
so here the idea is that you need to confirm that you want to merge if it is not a single commit because most of the time we want to squash the commits in a PR before merging, but it might be the case that we want separate commits, so there need to be a confirmation but the script should not just fail on it.

Reply:
To test: 
if you have 2 commits in a PR, script should ask for confirm
if you have 1 commit, it should just do the merge

Reply:
Got it. Will update soon.